crypto/evp/evp_lib.c

```diff
@@ -509,12 +509,17 @@ int EVP_CIPHER_CTX_get_iv_length(const EVP_CIPHER_CTX *ctx)
         size_t v = len;
         OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
 
-        params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_IVLEN, &v);
-        rv = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
-        if (rv != EVP_CTRL_RET_UNSUPPORTED) {
-            if (rv <= 0)
+        if (ctx->cipher->get_ctx_params != NULL) {
+            params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_IVLEN,
+                                                    &v);
+            rv = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
+            if (rv > 0) {
+                if (OSSL_PARAM_modified(params)
+                        && !OSSL_PARAM_get_int(params, &len))
```

Comment:
I think this is clearer than get_int() 
````
len = (size_t)v;
````

Reply:
That's one of the several problems being addressed here.

The cast has no range checking, whereas the `OSSL_PARAM_get_int()` call has full range checking.

Assuming you meant: `len = (int)v;`

Reply:
If we have a OSSL_PARAM that it defined as returning a size_t then it needs to return a size_t.. 
If it tried returning a -1 in a size_t then that is a provider side problem.. i.e. It should be returning an error.
I think this is addressing the problem in the wrong way.

Reply:
We're constrained by the existing API which returns an int.  It's a size so size_t makes perfect sense on the provider side but this function must return an int.  At some point the conversion needs to be done & it must be done safely (it's not just dealing with a -1 or -2 return).

The whole point of the OSSL_PARAM_get_ functions is to provide type conversion if required.  The value is fetched as a size_t a couple of lines above (line 515), this switches it to int without going to the provider again.
I really want to avoid lots of sequences like this:
```c
if (v > INT_MAX)
    return -1;
len = (int)v;
```
Such sequences are error prone and more difficult to maintain.